### PR TITLE
V washai/task module csharp

### PR DESCRIPTION
**Pull request by Wajeed-msft**
Added TaskModuleInfo, TaskModule & TaskModuleEnvelope model classes. Added test cases for the same.

### CODE REVIEW

File: CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleEnvelope.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// regenerated.
+
+namespace Microsoft.Bot.Connector.Teams.Models
+{
+    using System.Linq;
+
+    /// <summary>
+    /// Envelope for Task Module.
+    /// </summary>
+    public partial class TaskModuleEnvelope
```
Comment (siduppal) [inline CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleEnvelope.cs]: This represents the structure of the response to invoke for TaskModules. For ComposeExtensions, we used ComposeExtensionsResponse. Recommend to follow the same and rename to TaskModuleResponse.

File: CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleInfo.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+// regenerated.
+
+namespace Microsoft.Bot.Connector.Teams.Models
+{
+    using System.Linq;
+
+    /// <summary>
+    /// Metadata for a Task Module.
+    /// </summary>
+    public partial class TaskModuleInfo
```
Comment (siduppal) [inline CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleInfo.cs]: Need to rename this to describe what this class is about. "Info" doesn't convey much.

File: CSharp/Tests/Microsoft.Bot.Connector.Teams.Tests.Shared/TaskModuleTests.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            };
+
+            var expectedTaskModuleEnvelope = JsonConvert.DeserializeObject<TaskModuleEnvelope>(File.ReadAllText(@"Jsons\SampleTaskModuleMessagePayload.json"));
+            Assert.IsTrue(expectedTaskModuleEnvelope != null);
+            Assert.IsTrue(expectedTaskModuleEnvelope.Task.Type == currentTaskModuleEnvelope.Task.Type);
+            Assert.IsTrue(expectedTaskModuleEnvelope.Task.Value.ToString() == currentTaskModuleEnvelope.Task.Value.ToString());
+
+        }
+    }
+}
```
Comment (siduppal) [inline CSharp/Tests/Microsoft.Bot.Connector.Teams.Tests.Shared/TaskModuleTests.cs]: Please modify the TestBot to return a TaskModule response

File: CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleInfo.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (robin-liao) [reply CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleInfo.cs]: It looks like it's correct this class should name in `TaskInfo` or `TaskModuleInfo`. It's the type of "value" for "continue" type Task Module response.

[the remark this answers is not in the document]
Comment (robin-liao) [reply CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleInfo.cs]: FYI - It's called [`ITaskModuleTaskInfo` on Node.js side ](https://github.com/OfficeDev/BotBuilder-MicrosoftTeams/blob/ef06fa30cfbc962c575590ddf11532c6d8584f35/Node/src/models/TaskModuleResponse.ts#L49).

[the remark this answers is not in the document]
Comment (robin-liao) [reply CSharp/Library/Microsoft.Bot.Connector.Teams.Shared/TeamsAPI/Models/TaskModuleInfo.cs]: @Wajeed-msft  - I would vote yes for consistency.

File: CSharp/Tests/Microsoft.Bot.Connector.Teams.Tests.Shared/TaskModuleTests.cs (hunk 3 above)
Comment (Wajeed-msft) [reply CSharp/Tests/Microsoft.Bot.Connector.Teams.Tests.Shared/TaskModuleTests.cs]: @siduppal, @robin-liao   - Therse are the Task Module response classes which are returned as payload in response like _Request.CreateResponse(HttpStatusCode.OK, taskEnvelope);_

Do you have any test cased which I could use as reference?

**Pull request merged**